statamic/spock
# Simplify and listen to more data/content related events

**Goal**

The goal of this PR is to get Spock to listen to more /cp events.  Not concerned with changes made via Workshop at this point, but might be worth looking into after this is released.

**Prerequisites**

- [x] Simplify adding events to spock
- [x] Move pathing logic to statamic

**Content Section**

- [x] Listen on delete page
- [x] Listen on delete entry
- [x] Listen on delete term
- [x] Listen on edit global variable
- [x] Listen on upload asset
- [x] Listen on edit asset
- [x] Listen on delete asset
- [x] Listen on delete asset with meta data
- [x] Listen on move asset
- [x] Listen on rename asset
- [x] Listen on save asset folder
- [x] Listen on save asset folder with custom title
- [x] Listen on delete asset folder with custom title
- [x] Listen on delete asset folder containing images
- [x] Listen on save form submission
- [x] Listen on save form submission with file uploads
- [x] Listen on delete form submission
- [x] Listen when re-ordering or nesting pages in page tree

**Configure Section** (maybe? look into if easy enough)

- [x] Listen on content config (assets)
- [x] Listen on content config (collections)
- [x] Listen on content config (taxonomies)
- [x] Listen on content config (globals)
- [x] Listen on save fieldset
- [x] Listen on delete fieldset
- [x] Listen on save settings
- [x] Listen on save user
- [x] Listen on delete user
- [x] Listen on save user group
- [x] Listen on delete user group
- [x] Listen on save user role
- [x] Listen on delete user role

**How to test**

- Setup a test app.
- Setup Spock to git commit (see updated readme).
- Try messing with anything/everything in control panel.  Hopefully everything gets committed, and nothing is left unstaged!

**Prerelease**

- [ ] After the changes in Statamic are released and tagged, we'll need to add a little blurb to Spock's readme about older versions.  Maybe something like...

>If you intend to run Spock on a version of statamic older than ___, please use Spock version ___.
